### PR TITLE
feat: :loud_sound: add runtime logging for easier debugging

### PR DESCRIPTION
**Pull request by 25huizengek1**
Redirect JVM logging and log Minecraft version for easier debugging and support

### CODE REVIEW

**Review by 770grappenmaker — CHANGES_REQUESTED** [review]
LGTM. I'll ask @RichardDorian to merge. Though, you might want to reduce the repetitiveness and add a prefix, something like `[GAME]`.

**Comment by 25huizengek1** [comment]
I found a little issue though: if you set it to close the launcher after the game is launched, this won't really work. I don't know what to do with that, but I think it's not a big deal.

File: src/javascript/minecraft.js [hunk 1]
```diff
@@ -640,12 +641,16 @@ export async function launchGame(metadata, serverIp = null, debug = false) {
...
-      await settings.get('version')
+      version
     ),
     detached: true,
     shell: debug,
   });
 
+  logger.info(`Launching Minecraft ${version}`);
+  proc.stdout.addListener('data', data => logger.debug(data.toString("utf8")));
+  proc.stderr.addListener('data', data => logger.error(data.toString("utf8")));
```
Comment (dtomvan) [inline src/javascript/minecraft.js:652]: Is it guaranteed that `stdout` only returns debug logs, and `stderr` only errors?

I would redirect the JVM's `stdout` and `stderr` directly to the launcher's logger, maybe including an aforementioned `[GAME]` prefix, without assuming that specific output streams contain specific data.

You can prepend to each line with a [prepend-transform](https://github.com/ORESoftware/prepend-transform).

This way logging works by opening `latest.log` with `fs.createWriteStream`.

You get returned a file handle, which you can pipe the data to, while prepending `"[GAME]"`.

Example:
```js
const fs = require('fs');
import pt from 'prepend-transform';

const logFile = fs.createWriteStream("/path/to/latest.log");
const prepender = pt("[GAME] ");

proc.stdout.pipe(prepender).pipe(logFile);
proc.stderr.pipe(prepender).pipe(logFile);
```

This also avoids explicitly writing to the logfile as done in src/javascript/logger.js. This is an optimization, as the game will be sending logs frequently.

Comment (dtomvan) [reply src/javascript/minecraft.js:652]: You can pipe to the console by adding the lines:
```js
proc.stdout.pipe(process.stdout);
proc.stderr.pipe(process.stderr);
```

Comment (770grappenmaker) [reply src/javascript/minecraft.js:652]: @dtomvan According to the docs:
> Writes may be synchronous depending on what the stream is connected to and whether the system is Windows or POSIX:
Files: synchronous on Windows and POSIX
TTYs (Terminals): asynchronous on Windows, synchronous on POSIX
Pipes (and sockets): synchronous on Windows, asynchronous on POSIX

Source: https://nodejs.org/api/process.html#a-note-on-process-io  
The problem here is that pipes and file writes are very inconsistent; this means that usage of pipes is probably less consistent than listening for data and writing to the file.

**Comment by 770grappenmaker** [comment]
Closed in 6e6b2ff6a63a86a808a4657390cd35ca15a6f4ec

**Pull request closed without merging**